### PR TITLE
Display help output on stdout #8980

### PR DESCRIPTION
**Pull request by miminar**
This is a small adjustment of the work done by @rhatdan and @tiborvass. It
suppresses redundant error messages and handles a case where docker is run with
a bad option without a command.

Fixes #8980 and closes #7887

### CODE REVIEW

File: pkg/mflag/flag.go [hunk 1]
```diff
@@ -788,8 +851,12 @@ func Var(value Value, names []string, usage string) {
...
 func (f *FlagSet) failf(format string, a ...interface{}) error {
 	err := fmt.Errorf(format, a...)
-	fmt.Fprintln(f.out(), err)
-	f.usage()
+	fmt.Fprintln(f.Out(), err)
+	if os.Args[0] == f.name {
+		fmt.Fprintf(f.Out(), "See '%s --help'.\n", os.Args[0])
+	} else {
+		fmt.Fprintf(f.Out(), "See '%s %s --help'.\n", os.Args[0], f.name)
+	}
```
Comment (rhatdan) [inline pkg/mflag/flag.go]: I totally disagree.

We should follow the coreutils standard

ls --fobar
ls: unrecognized option '--fobar'
Try 'ls --help' for more information.

Only display usage if user asks for it.

**Comment by miminar** [comment]
@jfrazelle: Rebase done. For the last time I hope...
Thanks for taking care of it.

**Comment by rhatdan** [comment]
We have only been working on this one for 9 months now...

**Comment by jfrazelle** [comment]
great LGTM, ping @tiborvass :)

**Comment by tiborvass** [comment]
LGTM

If we discover other inconsistencies, we can fix them in a separate PR. I agree this took way longer than all of us wanted. Thanks everyone for your patience.

**Pull request merged**